pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -218,7 +233,7 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req Request) (Response
 			}
 
 			// Skip caching if the request is not cachable.
-			if !isRequestCachable(splitReq.orig, maxCacheTime, s.cacheUnalignedRequests, s.logger) {
+			if !isRequestCachable(splitReq.orig, maxCacheTime, s.cacheUnalignedRequests, s.logger, s.metrics.notCachableReason) {
```

Comment:
As you can see, `isRequestCachable()` is called in two places here, thus leading to double counting under some conditions. We shouldn't track it again here (but I'm wondering why wasn't spot in unit tests?).

Reply:
It did. It was counted twice and I fixed it.